core/util/proxy.py

```diff
@@ -0,0 +1,115 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+#import self.framework.requests
+from bs4 import BeautifulSoup
+import time
+#Python proxy module. Contributed by: @anuran-roy under @saeeddhqan
+class main:
+
+    def __init__(self, verbose = False):
+        '''
+        use : Determine whether to use proxy or not.
+        '''
+        self.use=self._global_options_['autoproxy']
+        self.verbose = verbose
+        self.iplist = []
+
+    def getproxy(self):
+        if self.use == True:
+            self.framework.output("[PROXY] Gathering proxies. It's quite time consuming, so do some pushups in the meantime :P")
+            URL = [ 'https://premproxy.com/list/' ]
+            i = 2
+            while i<=8:
```

Comment:
while i <= 8:

Reply:
I guess that this is important. Premproxy has only about 8 pages. Hence the limit @saeeddhqan . 😅